### PR TITLE
Support Rails 5.1

### PR DESCRIPTION
**Pull request by maschwenk**
Rails 5.1 release has been officially cut http://weblog.rubyonrails.org/2017/4/27/Rails-5-1-final/

Blocked by https://github.com/plataformatec/has_scope/pull/88

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ gemfile:
   - test/gemfiles/Gemfile-Rails-4-1
   - test/gemfiles/Gemfile-Rails-4-2
   - test/gemfiles/Gemfile-Rails-5-0
+  - test/gemfiles/Gemfile-Rails-5-1
 matrix:
   exclude:
     - rvm: 2.1.9
```
Comment (varyonic) [inline .travis.yml:17]: Needs
```
    - rvm: 2.1.9
      gemfile: test/gemfiles/Gemfile-Rails-5-1
```

**Review by varyonic — CHANGES_REQUESTED** [review]

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# Version 1.8.0
```
Comment (timoschilling) [inline CHANGELOG]: The dependency relax isn't a minor change, so we will release it under `1.7.2`

File: lib/inherited_resources/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module InheritedResources
-  VERSION = '1.7.1'.freeze
+  VERSION = '1.8.0'.freeze
```
Comment (timoschilling) [inline lib/inherited_resources/version.rb]: please don't make a version bump in a PR

File: CHANGELOG (hunk 2 above)
Comment (maschwenk) [reply CHANGELOG]: I assume you mean `major` change. Also, should I leave the changelog alone like the `version.rb` below? Just keep the actual code changes?

**Review by rafaelfranca — APPROVED** [review]

File: inherited_resources.gemspec [hunk 4]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.1'
 
   s.add_dependency("responders")
-  s.add_dependency("actionpack", ">= 3.2", "< 5.1")
-  s.add_dependency("railties", ">= 3.2", "< 5.1")
+  s.add_dependency("actionpack", ">= 3.2", "< 5.2")
```
Comment (varyonic) [inline inherited_resources.gemspec]: Should this be "< 5.2.x"?  < 5.1 was compatible with 5.1.rc2 but not 5.1.0 which was why we did not update sooner.

**Review by varyonic — APPROVED** [review]

File: inherited_resources.gemspec (hunk 4 above)
Comment (rafaelfranca) [reply inherited_resources.gemspec]: :+1:

**Comment by rafaelfranca** [comment]
has_scope 0.7.1 was released

**Comment by deivid-rodriguez** [comment]
I restarted the build and it installed fine, but the tests didn't pass. So it seems that there's actual work to be done here... :(

**Comment by maschwenk** [comment]
Ahh yes. Looks like the Rails 5.0 change about putting all params in a params hash. Will fix when home.

File: inherited_resources.gemspec [hunk 5]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.1'
 
   s.add_dependency("responders")
-  s.add_dependency("actionpack", ">= 3.2", "< 5.1")
-  s.add_dependency("railties", ">= 3.2", "< 5.1")
+  s.add_dependency("actionpack", ">= 3.2", "< 5.2.x")
+  s.add_dependency("railties", ">= 3.2", "< 5.2.x")
```
Comment (Empact) [inline inherited_resources.gemspec:24]: I don't think you want `.x` here.

Comment (maschwenk) [reply inherited_resources.gemspec:24]: see https://github.com/activeadmin/inherited_resources/pull/433#discussion_r114039528

**Comment by maschwenk** [comment]
Phew! more involved than I had suspected. I'm pretty new to working with this backwards compatibility stuff. Now I understand why Gem owners drop legacy Rails support 😉

**Review by varyonic — APPROVED** [review]

**Comment by deivid-rodriguez** [comment]
Fantastic work @maschwenk, I don't have time to look at the diff right now, but the green checks look great! :)

**Review by deivid-rodriguez — APPROVED** [review]
Had a quick look and looks good to me. Not really aware of all the test class changes in https://github.com/activeadmin/inherited_resources/pull/433/commits/34d8fde5b3d6cee676660f5040935b53d2ee3c69, but they seem to be confirmed by the passing build. In any case, I'm 100% sure Rafael knows this stuff and can confirm :)

**Review by rafaelfranca — APPROVED** [review]

**Comment by deivid-rodriguez** [comment]
Went ahead and merged this. Only pending a release! :tada:

**Comment by varyonic** [comment]
Thanks @deivid-rodriguez Bump version and `rake release` is down to @rafaelfranca or @timoschilling

**Comment by timoschilling** [comment]
I will do it

**Pull request merged**